src/ansys/mechanical/core/embedding/app.py

```diff
@@ -236,6 +236,13 @@ def execute_script(self, script: str) -> typing.Any:
             raise Exception(error_msg)
         return script_result.Value
 
+    def execute_script_from_file(self, file_path=None):
+        """Execute the given script from file with the internal IronPython engine."""
+        text_file = open(file_path, "r", encoding="utf-8")
```

Comment:
any exceptions raised in the file won't have the filename anymore. Maybe that's ok.